x/rollapp/abci.go

```diff
@@ -17,41 +17,57 @@ func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper)
 
 // Called every block to finalize states that their dispute period over.
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
+
 	// check to see if there are pending  states to be finalized
-	blockHeightToFinalizationQueue, found := k.GetBlockHeightToFinalizationQueue(ctx, uint64(ctx.BlockHeight()))
-	if !found {
-		return
+	//blockHeightToFinalizationQueue, found := k.GetBlockHeightToFinalizationQueue(ctx, uint64(ctx.BlockHeight()))
+	globalIndex, found := k.GetLatestFinalizedStateGlobalIndex(ctx)
+
+	finalizationHeight := uint64(0)
+	if uint64(ctx.BlockHeight()) >= k.DisputePeriodInBlocks(ctx) {
```

Comment:
why are we just not returning if this condition is not met?

Reply:
outdated code